Task2/stack.c

```diff
@@ -0,0 +1,104 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include "stack.h"
+
+int createStack(Stack** stackPtrPtr)
+{
+    *stackPtrPtr = calloc(1, sizeof(Stack));
+    if (*stackPtrPtr == NULL)
+    {
+        return -1;
+    }
+    return 0;
+}
+
+int push(Stack* stackPtr, int value)
+{
+    if (stackPtr == NULL)
+    {
+        return -2;
+    }
+    StackElement* newElementPtr = calloc(1, sizeof(StackElement));
+    if (newElementPtr == NULL)
+    {
+        return -1;
+    }
+    newElementPtr->nextElementPtr = stackPtr->lastElementPtr;
+    newElementPtr->value = value;
+    stackPtr->lastElementPtr = newElementPtr;
+    return 0;
+}
+
+int pop(Stack* stackPtr, int* value)
+{
+    if (stackPtr == NULL || stackPtr->lastElementPtr == NULL)
+    {
+        return -1;
+    }
+    *value = stackPtr->lastElementPtr->value;
+    StackElement* PtrTolastElementToFree = stackPtr->lastElementPtr;
+    stackPtr->lastElementPtr = stackPtr->lastElementPtr->nextElementPtr;
+    free(PtrTolastElementToFree);
+    return 0;
+}
+
+int add(Stack* stackPtr)
+{
+    if (stackPtr == NULL)
+    {
+        return -2;
+    }
+    int firstNumber = 0;
+    int secondNumber = 0;
+    int firstErrorCode = pop(stackPtr, &firstNumber);
+    if (firstErrorCode == -1)
+    {
+        return -3;
+    }
+    int secondErrorCode = pop(stackPtr, &secondNumber);
+    if (secondErrorCode == -1)
+    {
+        push(stackPtr, firstNumber);
+        return -3;
+    }
+    push(stackPtr, firstNumber + secondNumber);
+    return 0;
+}
+
+int printStack(Stack* stackPtr)
+{
+    if (stackPtr == NULL)
+    {
+        return -1;
+    }
+    StackElement* currentElementPtr = stackPtr->lastElementPtr;
+    if (currentElementPtr == NULL)
+    {
+        printf("Stack is empty. \n");
```

Comment:
```suggestion
        printf("Stack is empty. \n");
        retun 0;
```
Если стек пуст, то не нужно его печатать.
Ибо получается странный вывод в консоль
```
Stack is empty.
Stack: 
```
(-0.5 балла)